src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java

```diff
@@ -1,31 +1,60 @@
 package thewaypointers.trafficsimulator;
+import javax.swing.JButton;
 
+import thewaypointers.trafficsimulator.common.IStateProvider;
 import thewaypointers.trafficsimulator.common.WorldStateDTO;
 import thewaypointers.trafficsimulator.gui.MainFrame;
-import thewaypointers.trafficsimulator.simulation.Simulation;
 
-public class TrafficSimulatorManager {
+public class TrafficSimulatorManager extends Thread {
 
-    final static long TIME_STEP = 100;
-    final static long SIMULATION_TIME_STEP = 300;
-    static WorldStateDTO worldState;
-    static MainFrame mainFrame;
+    public static final float VEHICLE_MOVEMENT_SPEED = 2;
+    public static final float STATES_PER_SECOND = 1;
+    private static MainFrame mainFrame;
+    private static IStateProvider simulation;
+    private WorldStateDTO worldState;
+    private boolean stop = true;
+    private boolean isSleep = true;
```

Comment:
You should synchronize access to those control variables since they will be used by different threads. If you don't, it might not work at all, since due to how Java's memory model works the new values might get written to local caches of the threads and not really get shared between them. In this case marking with [volatile](http://www.javamex.com/tutorials/synchronization_volatile.shtml) should be sufficient.